MetaLamp/nft-marketplace/src/Ext/Plutus/Contracts/Auction.hs

```diff
@@ -16,36 +16,46 @@
-import           Ledger.Constraints.TxConstraints (TxConstraints)
-import qualified Ledger.Interval                  as Interval
-import qualified Ledger.Typed.Scripts             as Scripts
-import           Ledger.Typed.Tx                  (TypedScriptTxOut (..))
-import           Ledger.Value                     (AssetClass)
+import qualified Ledger.Ada                                       as Ada
+import qualified Ledger.Constraints                               as Constraints
+import           Ledger.Constraints.TxConstraints                 (TxConstraints)
+import qualified Ledger.Interval                                  as Interval
+import qualified Ledger.Typed.Scripts                             as Scripts
+import           Ledger.Typed.Tx                                  (TypedScriptTxOut (..))
+import           Ledger.Value                                     (AssetClass)
 import           Plutus.Contract
-import           Plutus.Contract.StateMachine     hiding (mkValidator,
-                                                   typedValidator)
-import qualified Plutus.Contract.StateMachine     as SM
-import           Plutus.Contract.Util             (loopM)
-import qualified Plutus.Contracts.Currency        as Currency
+import           Plutus.Contract.StateMachine                     hiding
+                                                                  (mkValidator,
+                                                                   typedValidator)
+import qualified Plutus.Contract.StateMachine                     as SM
+import           Plutus.Contract.Util                             (loopM)
+import qualified Plutus.Contracts.Currency                        as Currency
+import qualified Plutus.Contracts.NftMarketplace.OnChain.Core.NFT as Core
+import           Plutus.Types.Percentage                          (Percentage (..))
 import qualified PlutusTx
 import           PlutusTx.Prelude
-import qualified Prelude                          as Haskell
+import qualified PlutusTx.Ratio                                   as Ratio
+import qualified Prelude                                          as Haskell
 
 -- | Definition of an auction
 data AuctionParams
     = AuctionParams
-        { apOwner   :: PubKeyHash -- ^ Current owner of the asset. This is where the proceeds of the auction will be sent.
-        , apAsset   :: Value -- ^ The asset itself. This value is going to be locked by the auction script output.
-        , apEndTime :: Ledger.POSIXTime -- ^ When the time window for bidding ends.
+        { apOwner               :: PubKeyHash -- ^ Current owner of the asset. This is where the proceeds of the auction will be sent.
+        , apAsset               :: Value -- ^ The asset itself. This value is going to be locked by the auction script output.
+        , apEndTime             :: Ledger.POSIXTime -- ^ When the time window for bidding ends.
+        , apInitialPrice        :: Value
```

Comment:
Currently this auction only supports bids in Ada , so I think this field should be of the same type as `highestBid    :: Ada` .

We could just delete the field for now if it is not used.